### DISCUSSION TITLE
Threadsafe

### DISCUSSION THREAD

**Discussion by zac-wang-nv** [snakers4/silero-vad#305] [Q&A] [answered]
First, thanks for this great work. I am in the process of integrating it to our stt system.
I would like to use the get_speech_timestamps api in a multi-threading case, and I find that the function internally calls
`model.reset_states()` at the beginning, is this thread-safe?  i haven't got a chance to do more detailed testing on this yet. Any insights on this matter will be very helpful.  Thanks in advance!

**Comment by snakers4** [comment, marked as answer]
I believe the proper way is to have a separate instance of VAD for each process / thread. Or to have a separate process / process pool executor with it.

**Reply by snakers4** [reply]
https://github.com/snakers4/silero-vad/issues/2#issuecomment-1050968656

**Comment by zac-wang-nv** [comment]
Thank you for the reply, I manage to get it to work with a process pool executor.

**Reply by snakers4** [reply]
Can you maybe share a minimalist example in the examples folder?
